middle_end/flambda2/simplify/simplify_unary_primitive.ml

```diff
@@ -168,13 +170,14 @@ let simplify_box_number (boxable_number_kind : K.Boxable_number.t) dacc
     | Untagged_immediate -> T.tag_immediate naked_number_ty
   in
   let dacc = DA.add_variable dacc result_var ty in
-  Simplified_named.reachable original_term ~try_reify:true, dacc
+  Simplified_named.reachable original_term ~try_reify:(Some ty), dacc
 
 let simplify_is_int_or_get_tag dacc ~original_term ~scrutinee ~scrutinee_ty:_
     ~result_var ~make_shape =
   (* CR mshinwell: Check [scrutinee_ty] (e.g. its kind)? *)
+  let ty = make_shape scrutinee in
   let dacc = DA.add_variable dacc result_var (make_shape scrutinee) in
```

Comment:
You should probably re-use `ty` here instead of re-computing `make_shape scrutinee`